app/scripts/background.js

```diff
@@ -720,10 +721,18 @@ function setupController(initState, initLangCode) {
           REJECT_NOTFICIATION_CLOSE,
         ),
       );
-
-    // Finally, reject all approvals managed by the ApprovalController
-    controller.approvalController.clear(
-      ethErrors.provider.userRejectedRequest(),
+    // Finally, resolve snap dialog approvals and reject all the others managed by the ApprovalController
+    Object.values(controller.approvalController.state.pendingApprovals).forEach(
```

Comment:
This should probably be fenced?

Reply:
How could we fence this so it also works when it's not Flask ?